apps/web/src/components/docs/Docs.tsx

```diff
@@ -0,0 +1,363 @@
+import { colors, IconInfoOutline, IconOutlineWarning, Tabs, Tooltip } from '@novu/design-system';
+import { Accordion, Alert, Code, Loader, Paper } from '@mantine/core';
+import { PropsWithChildren, ReactNode, useEffect, useMemo } from 'react';
+import * as mdxBundler from 'mdx-bundler/client';
+import { Highlight } from './Highlight';
+import { useQuery } from '@tanstack/react-query';
+import { useSegment } from '@novu/shared-web';
+import { Center, Flex, Grid, GridItem, styled, VStack } from '../../styled-system/jsx';
+import { css } from '../../styled-system/css';
+import { text, title as RTitle } from '../../styled-system/recipes';
+import { DOCS_URL, MDX_URL, MINTLIFY_IMAGE_URL } from './docs.const';
+
+const Text = styled('p', text);
+const LiText = styled('span', text);
+const TitleH2 = styled('h2', RTitle);
+const TitleH3 = styled('h3', RTitle);
+const TitleH1 = styled('h1', RTitle);
+
+const getMDXComponent = mdxBundler.getMDXComponent;
+
+/*
+ *Render the mdx for our mintlify docs inside of the web.
+ *Fetching the compiled mdx from another service and then try to map the markdown to react components.
+ */
+export const Docs = ({ path = '', children, actions }: PropsWithChildren<{ path?: string; actions: ReactNode }>) => {
+  const segment = useSegment();
+
+  const { isLoading, data: { code, title, description } = { code: '', title: '', description: '' } } = useQuery<{
+    code: string;
+    title: string;
+    description: string;
+  }>(['docs', path], async () => {
+    const response = await fetch(MDX_URL + path);
+    const json = await response.json();
+
+    return json;
+  });
+
+  useEffect(() => {
+    segment.track('Inline docs opened', {
+      documentationPage: path,
+      pageURL: window.location.href,
+    });
+  }, [path, segment]);
+
+  const Component = useMemo(() => {
+    if (code.length === 0) {
+      return null;
+    }
+
+    return getMDXComponent(code);
+  }, [code]);
+
+  // Workaround for img tags that is not parsed correctly by mdx-bundler
+  useEffect(() => {
+    if (Component === null || isLoading) {
+      return;
+    }
+
+    const docs = document.getElementById('docs');
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** I wonder if we should have a more specific `id` here, `docs` could be a little loose and result in multiple elements being targeted. I'd suggest something like `embedded-docs`

Furthermore, let's extract this out to a constant.‏